### PR TITLE
Add JT command to print environment

### PR DESCRIPTION
**Pull request by bjfish**
This could be used to help when debugging issues from contributors/users.

Example output: https://gist.github.com/bjfish/b231feba123200d50581d0ac2f519f80

### CODE REVIEW

File: tool/jt.rb [hunk 1]
```diff
@@ -391,6 +391,7 @@ def help
           options                                    build the options
       jt build_stats [--json] <attribute>            prints attribute's value from build process (e.g., binary size)
       jt clean                                       clean
+      jt environment                                 prints the current environment
```
Comment (eregon) [inline tool/jt.rb]: `environment` is long and not so easy to type, I think we should abbreviate it to `jt env`

File: tool/jt.rb [hunk 2]
```diff
@@ -517,6 +518,26 @@ def dis(file)
...
+  def environment
+    puts "Environment"
+    env_vars = ['JAVA_HOME','RUBY_BIN','GRAALVM_BIN','GRAAL_HOME',
+    'JVMCI_BIN','JVMCI_GRAAL_HOME','GRAAL_JS_JAR',
+    'SL_JAR','OPENSSL_PREFIX','AOT_BIN','TRUFFLERUBY_CEXT_ENABLED',
+    'TRUFFLERUBYOPT','RUBYOPT']
+    env_vars.each do |e|
+      puts "#{e}: #{ENV[e]}"
+    end
+    raw_sh('ruby -v', continue_on_failure: true)
```
Comment (eregon) [inline tool/jt.rb]: You have to split those arguments:
```ruby
raw_sh('ruby', '-v', continue_on_failure: true)
```

You could create a helper lambda in this method if want:
```ruby
shell = -> command { raw_sh(*command.split, continue_on_failure: true) }
shell['ruby -v'] # or .call()
```

File: tool/jt.rb [hunk 3]
```diff
@@ -517,6 +518,26 @@ def dis(file)
     puts Pathname(file).sub_ext('.ll')
   end
 
+  def environment
+    puts "Environment"
+    env_vars = ['JAVA_HOME','RUBY_BIN','GRAALVM_BIN','GRAAL_HOME',
+    'JVMCI_BIN','JVMCI_GRAAL_HOME','GRAAL_JS_JAR',
+    'SL_JAR','OPENSSL_PREFIX','AOT_BIN','TRUFFLERUBY_CEXT_ENABLED',
+    'TRUFFLERUBYOPT','RUBYOPT']
```
Comment (eregon) [inline tool/jt.rb]: I would like to have `PATH` in there, it's often the source of problems.

File: tool/jt.rb [hunk 4]
```diff
@@ -517,6 +518,26 @@ def dis(file)
...
+    env_vars.each do |e|
+      puts "#{e}: #{ENV[e]}"
```
Comment (eregon) [inline tool/jt.rb]: `#{ENV[e].inspect}` so we can differentiate empty string and unset variable.

File: tool/jt.rb [hunk 5]
```diff
@@ -517,6 +518,26 @@ def dis(file)
...
+    end
+    raw_sh('ruby -v', continue_on_failure: true)
+    raw_sh('uname -a', continue_on_failure: true)
+    raw_sh('clang -v', continue_on_failure: true)
+    raw_sh('opt -version', continue_on_failure: true)
+    raw_sh('mx version', continue_on_failure: true)
+    sh('mx sversions', continue_on_failure: true)
+    raw_sh('git show -s --format=%H', continue_on_failure: true)
+    raw_sh("#{ENV['OPENSSL_PREFIX']}/bin/openssl version", continue_on_failure: true)
```
Comment (eregon) [inline tool/jt.rb]: This should also work if OPENSSL_PREFIX is not set (e.g. on Linux)

File: tool/jt.rb (hunk 3 above)
Comment (eregon) [reply tool/jt.rb]: Also `TRUFFLERUBY_RESILIENT_GEM_HOME`

Comment (eregon) [reply tool/jt.rb]: BTW, `%w[foo bar]` can be nicely used here to avoid quotes and commas.

File: tool/jt.rb (hunk 1 above)
Comment (bjfish) [reply tool/jt.rb]: Fixed.

File: tool/jt.rb (hunk 3 above)
Comment (bjfish) [reply tool/jt.rb]: Fixed all these.

File: tool/jt.rb (hunk 4 above)
Comment (bjfish) [reply tool/jt.rb]: Fixed.

File: tool/jt.rb (hunk 2 above)
Comment (bjfish) [reply tool/jt.rb]: Fixed. Used the lambda. Thanks.

File: tool/jt.rb (hunk 5 above)
Comment (bjfish) [reply tool/jt.rb]: Fixed

File: tool/jt.rb [hunk 6]
```diff
@@ -517,6 +518,32 @@ def dis(file)
...
+    env_vars.each do |e|
+      puts "#{e}: #{ENV[e].inspect}"
+    end
+    shell = -> command { raw_sh(*command.split, continue_on_failure: true) }
+    shell['ruby -v']
+    shell['uname -a']
+    shell['clang -v']
+    shell['opt -version']
+    shell['mx version']
+    sh('mx','sversions', continue_on_failure: true)
```
Comment (eregon) [inline tool/jt.rb]: Minor, missing space after the comma here

File: tool/jt.rb [hunk 7]
```diff
@@ -517,6 +518,32 @@ def dis(file)
...
+    shell['git show -s --format=%H']
```
Comment (eregon) [inline tool/jt.rb]: This shows the result in a pager at me.

Comment (eregon) [reply tool/jt.rb]: One way to fix it is to use:
```
git --no-pager show -s --format=%H
```

Comment (eregon) [reply tool/jt.rb]: I fixed this and the one above.

**Pull request merged**